src/transformers/models/donut/feature_extraction_donut.py

```diff
@@ -0,0 +1,207 @@
+            `PIL.Image.BILINEAR`, `PIL.Image.HAMMING`, `PIL.Image.BICUBIC` or `PIL.Image.LANCZOS`. Only has an effect
+            if `do_resize` is set to `True`.
+        do_align_long_axis (`bool`, *optional*, defaults to `False`):
+            Whether to rotate the input if the height is greater than width.
+        do_pad (`bool`, *optional*, defaults to `True`):
+            Whether or not to pad the input to `size`.
+        do_normalize (`bool`, *optional*, defaults to `True`):
+            Whether or not to normalize the input with mean and standard deviation.
+        image_mean (`List[int]`, defaults to `[0.5, 0.5, 0.5]`):
+            The sequence of means for each channel, to be used when normalizing images.
+        image_std (`List[int]`, defaults to `[0.5, 0.5, 0.5]`):
+            The sequence of standard deviations for each channel, to be used when normalizing images.
+
+    """
+
+    model_input_names = ["pixel_values"]
+
+    def __init__(
+        self,
+        do_resize=True,
+        size=[1920, 2560],
+        resample=Image.BILINEAR,
+        do_align_long_axis=False,
+        do_pad=True,
+        do_normalize=True,
+        image_mean=None,
+        image_std=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.do_resize = do_resize
+        self.size = size
+        self.resample = resample
+        self.do_align_long_axis = do_align_long_axis
+        self.do_pad = do_pad
+        self.do_normalize = do_normalize
+        self.image_mean = image_mean if image_mean is not None else IMAGENET_STANDARD_MEAN
+        self.image_std = image_std if image_std is not None else IMAGENET_STANDARD_STD
+
+    def rotate(self, image, size):
+        if not isinstance(image, Image.Image):
+            image = self.to_pil_image(image)
+
+        if (size[1] > size[0] and image.width > image.height) or (size[1] < size[0] and image.width < image.height):
+            image = image.rotate(angle=-90, expand=True)
+
+        return image
+
+    def resize_and_thumbnail(self, image, size, resample):
+        # 1. resize the shorter edge of the image to `min(size)`
+        image = self.resize(image, size=min(size), resample=resample, default_to_square=False)
+        # 2. create a thumbnail
+        image.thumbnail((size[0], size[1]))
+
+        return image
+
+    def pad(self, image: Image.Image, size: Tuple[int, int], random_padding: bool = False) -> Image.Image:
+        delta_width = size[0] - image.width
+        delta_height = size[1] - image.height
+
+        if random_padding:
+            pad_width = np.random.randint(low=0, high=delta_width + 1)
+            pad_height = np.random.randint(low=0, high=delta_height + 1)
+        else:
+            pad_width = delta_width // 2
+            pad_height = delta_height // 2
+
+        padding = (
+            pad_width,
+            pad_height,
+            delta_width - pad_width,
+            delta_height - pad_height,
+        )
```

Comment:
```suggestion
        padding = (pad_width, pad_height, delta_width - pad_width, delta_height - pad_height)
```
This fits in one line.